flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -81,25 +127,36 @@ public void configure(Configuration parameters) {
 	 * @throws IOException
 	 */
 	@Override
-	public void open(InputSplit ignored) throws IOException {
+	public void open(InputSplit inputSplit) throws IOException {
+		hasNext = true;
 		try {
-			establishConnection();
-			statement = dbConn.createStatement(resultSetType, resultSetConcurrency);
-			resultSet = statement.executeQuery(query);
+			if(inputSplit!=null && inputSplit instanceof QueryParamInputSplit) {
+				QueryParamInputSplit jdbcInputSplit = (QueryParamInputSplit) inputSplit;
+				for (int i = 0; i < jdbcInputSplit.getQueryParameters().length; i++) {
+					Object param = jdbcInputSplit.getQueryParameters()[i];
+					if(param instanceof String) {
+						statement.setString(i + 1, (String) param);
+					} else if(param instanceof Long) {
+						statement.setLong(i + 1, (Long) param);
+					} else if(param instanceof Integer) {
+						statement.setInt(i + 1, (Integer) param);
+					} else if(param instanceof Double) {
+						statement.setDouble(i + 1, (Double) param);
+					} else if(param instanceof Boolean) {
+						statement.setBoolean(i + 1, (Boolean) param);
+					} else {
```

Comment:
Add common time types (data, time, timestamp) and the remaining primitives: byte, short, float, ...